### PR TITLE
add inline destructor

### PR DESCRIPTION
**Pull request by Iroy30**

### CODE REVIEW

File: cpp/include/raft/comms/comms.hpp [hunk 1]
```diff
@@ -323,7 +323,7 @@ class comms_t {
   std::unique_ptr<comms_iface> impl_;
 };
 
-comms_iface::~comms_iface() {}
+inline comms_iface::~comms_iface() {}
```
Comment (seunghwak) [inline cpp/include/raft/comms/comms.hpp:326]: Is there any reason in placing this outside the class definition while we keep the definition of a constructor inside the class definition? If there is no specific reason, we'd better keep this inside the class definition. And we're doing nothing special here, so ~comms_iface() = default; should be better. And are we allowing copy/move constructions/assignments or constructor without an argument? Based on that, better be explicit about either deleting them or marking those as default.

Comment (afender) [reply cpp/include/raft/comms/comms.hpp:326]: Making it `inline` has the same effect as placing it inside the class definition. 
Although while discussing the fix with Ishika I was also curious about the motivation for adding it outside in the first place.

Comment (teju85) [reply cpp/include/raft/comms/comms.hpp:326]: Also, if this is just a default dtor, why not just remove it altogether?

Comment (cjnolet) [reply cpp/include/raft/comms/comms.hpp:326]: Indeed, this should just removed.

Comment (Iroy30) [reply cpp/include/raft/comms/comms.hpp]: Thanks, I was not sure if this was a placeholder for something to come. Either way, I have deleted the same.

**Pull request merged**